### PR TITLE
Add CLI node info command

### PR DESCRIPTION
**Pull request by qrayven**
# Description of change

add `node-info` command that prints:

```shell
rayven@xhost:~/wallet-lab$ wallet node-info
The current node info: {
  "nodeInfo": {
    "name": "HORNET",
    "version": "2.0.0-rc.6",
    "status": {
      "isHealthy": true,
      "latestMilestone": {
        "index": 5689298,
        "timestamp": 1686928581,
        "milestoneId": "0x2528b160e7fbd2fa543dc94906761ef65f86b8354cef968fdc19344bcc353edf"
      },
      "confirmedMilestone": {
        "index": 5689298,
        "timestamp": 1686928581,
        "milestoneId": "0x2528b160e7fbd2fa543dc94906761ef65f86b8354cef968fdc19344bcc353edf"
      },
      "pruningIndex": 4669930
    },
    "supportedProtocolVersions": [
      2
    ],
    "protocol": {
      "version": 2,
      "networkName": "testnet-1",
      "bech32Hrp": "rms",
      "minPowScore": 1500,
      "belowMaxDepth": 15,
      "rentStructure": {
        "vByteCost": 100,
        "vByteFactorKey": 10,
        "vByteFactorData": 1
      },
      "tokenSupply": "1450896407249092"
    },
    "pendingProtocolParameters": [],
    "baseToken": {
      "name": "Shimmer",
      "tickerSymbol": "SMR",
      "unit": "SMR",
      "subunit": "glow",
      "decimals": 6,
      "useMetricPrefix": false
    },
    "metrics": {
      "blocksPerSecond": 3.0,
      "referencedBlocksPerSecond": 3.4,
      "referencedRate": 113.33333333333331
    },
    "features": []
  },
  "url": "https://api.testnet.shimmer.network"
}
```

## Links to any relevant issues

part of #389 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
I think it may be interesting to have it on both the Wallet and the Account

File: cli/src/command/wallet.rs [hunk 1]
```diff
@@ -196,6 +198,15 @@ pub async fn new_account_command(
     Ok((wallet, alias))
 }
 
+pub async fn node_info_command(storage_path: &Path, snapshot_path: &Path) -> Result<Wallet, Error> {
+    let password = get_password("Stronghold password", !snapshot_path.exists())?;
+    let wallet = unlock_wallet(storage_path, snapshot_path, password).await?;
+    let node_info = wallet.client().get_info().await?;
+
+    println_log_info!("The current node info: {}", serde_json::to_string_pretty(&node_info)?);
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs]:
```suggestion
    println_log_info!("The current node info: {}", serde_json::to_string_pretty(&node_info)?);

```

File: cli/src/command/wallet.rs [hunk 2]
```diff
@@ -196,6 +198,15 @@ pub async fn new_account_command(
     Ok((wallet, alias))
 }
 
+pub async fn node_info_command(storage_path: &Path, snapshot_path: &Path) -> Result<Wallet, Error> {
+    let password = get_password("Stronghold password", !snapshot_path.exists())?;
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs]: We don't really need the secret manager for this so it's a bit annoying to request a password. I would modify `unlock_wallet` to take an `Option<Password>` and conditionally create a secret manager. And pass None for this command.

Comment (qrayven) [reply cli/src/command/wallet.rs]: yeah, totally agree that is annoying. I didn't know that is possible, fixed.

**Comment by qrayven** [comment]
> I think it may be interesting to have it on both the Wallet and the Account

added

File: cli/src/command/account.rs [hunk 3]
```diff
@@ -595,6 +597,15 @@ pub async fn new_address_command(account: &Account) -> Result<(), Error> {
     Ok(())
 }
 
+// `node-info` command
+pub async fn node_info_command(account: &Account) -> Result<(), Error> {
+    let node_info = account.client().get_info().await?;
+
+    println_log_info!("The current node info: {}", serde_json::to_string_pretty(&node_info)?);
```
Comment (thibault-martinez) [inline cli/src/command/account.rs]:
```suggestion
    println_log_info!("Current node info: {}", serde_json::to_string_pretty(&node_info)?);
```

File: cli/src/command/wallet.rs [hunk 4]
```diff
@@ -196,6 +198,15 @@ pub async fn new_account_command(
     Ok((wallet, alias))
 }
 
+pub async fn node_info_command(storage_path: &Path) -> Result<Wallet, Error> {
+    let wallet = unlock_wallet(storage_path, None, None).await?;
+    let node_info = wallet.client().get_info().await?;
+
+    println_log_info!("The current node info: {}", serde_json::to_string_pretty(&node_info)?);
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs]:
```suggestion
    println_log_info!("Current node info: {}", serde_json::to_string_pretty(&node_info)?);
```

File: documentation/cli/docs/02-wallet.md [hunk 5]
```diff
@@ -179,6 +179,18 @@ Create a new account with a provided alias.
 ./wallet new-account main
 ```
 
+### `./wallet node-info`
+
+Displays information about currently set node.
+
+The wallet needs to be initialised (`init` command).
```
Comment (Thoralf-M) [inline documentation/cli/docs/02-wallet.md:186]: Do we want this here? I see it's also in `./wallet new-account`, but in many other commands not and don't we get the automatic init now if it wasn't initialized already?

Comment (qrayven) [reply documentation/cli/docs/02-wallet.md:186]: there is automatic `init` only when `wallet` without any additional subcommand is used. So technically is possible to enter the state when wallet is not initialized

File: cli/src/command/wallet.rs [hunk 6]
```diff
@@ -236,12 +247,22 @@ pub async fn sync_command(storage_path: &Path, snapshot_path: &Path) -> Result<W
...
-            .password(password)
-            .build(snapshot_path)?,
-    );
+pub async fn unlock_wallet(
+    storage_path: &Path,
+    snapshot_path: impl Into<Option<&Path>>,
+    password: impl Into<Option<Password>>,
+) -> Result<Wallet, Error> {
+    let secret_manager = if let Some(password) = password.into() {
+        let snapshot_path: Option<&Path> = snapshot_path.into();
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs]:
```suggestion
        let snapshot_path = snapshot_path.into();
```

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**